ledger/src/blockstore_processor.rs

```diff
@@ -1849,10 +1867,11 @@ fn load_frozen_forks(
                 entry_notification_sender,
                 None,
                 timing,
-            )
-            .is_err()
-            {
+            ) {
                 assert!(bank_forks.write().unwrap().remove(bank.slot()).is_some());
+                if opts.abort_on_invalid_block {
+                    Err(error)?
+                }
```

Comment:
this is needed for correct working of the new generic `--abort-on-invalid-block` flag, which in turn is used inside `run-sanity.sh`

Reply:
yeah. the total of 4 combination of `--block-verification-method` and `--abort-on-invalid-block` behaves consistently now.